pkg/xds/envoy/types.go

```diff
@@ -21,11 +23,118 @@ type ClusterSubset struct {
 }
 
 type Tags map[string]string
+type TagsSlice []Tags
+type TagKeys []string
+type TagKeysSlice []TagKeys
 
-func (t Tags) WithoutTag(tag string) Tags {
+func (t TagsSlice) ToTagKeySlice() TagKeysSlice {
+	out := []TagKeys{}
+	for _, v := range t {
+		out = append(out, v.Keys())
+	}
+	return out
+}
+
+// Transform applies each transformer to each TagSlice and return a sorted unique TagKeysSlice.
+func (t TagKeysSlice) Transform(transformers ...TagKeyTransformer) TagKeysSlice {
+	allSlices := map[uint64]TagKeys{}
+	for _, slice := range t {
+		res := slice.Transform(transformers...)
+		if len(res) > 0 {
+			allSlices[res.Hash()] = res
```

Comment:
This FNV hash algorithm has some collision rate, even though it's quite low, I'm still concerned about just rewriting the values with the same hash. If it shoots one day we will never debug it. I see 2 options here: 
- classic `equal` and `buckets`
- build hashes without collisions, tagKeys are not huge it will be fine to convert them to `string` and name it a hash

I like the second approach more since it simpler